source/includes/_ethereum.md

```diff
@@ -725,30 +902,148 @@ The random datasource is currently available on the Ethereum mainnet and on all
 
 <aside class="notice">
-The ProofShield is still EXPERIMENTAL, please DO NOT use it in production (yet). A production-ready version will follow in the future.
+Note that any calls to `oraclize_setCustomTokenPayment()` or `oraclize_getPriceERC20(...)'` that use a token-address not on the list of supported tokens will `revert();`.
 </aside>
 
-The ProofShield enables smart contracts to verify on-chain the authenticity proofs provided by Oraclize, this ensures that the authenticity of the data received is verified before going ahead and using the data.
+The same as when paying for queries with ETH, when paying with a token you will need to ensure your contract has a sufficient balance of that token to cover the query cost, and that a balance has been `approved` to be used by the Provable connector contract which it is calling. (See the ERC20 token __[specification here](https://theethereum.wiki/w/index.php/ERC20_Token_Standard#Approve_And_TransferFrom_Token_Balance)__ for more information on token approval.)
+
+To make this simpler, the Provable API provides various helper functions which you can see on the right hand side.
+
+<aside class="notice">
+Notice in the helper functions the one that composes both `oraclize_setCustomTokenPayment` and `oraclize_approveTokenAllowance`. This composition (`oraclize_setAndApproveTokenAllowance(...);`) is ideal for using in your contract's constructor function to setup the token payments in a single function call upon deployment!
+</aside>
+
+Just as when __[paying for queries using ETH](http://docs.oraclize.it/#ethereum-best-practices-pre-calculating-the-query-price)__, when paying using a token it is also possible to pre-calculate a query-price via: `oraclize_getPriceERC20(<datasource>);`. This returns the next query price expressed in units of the token that the your contract has set as its `customTokenPayment`.
 
+Additional parameters may be passed to get prices for more specific query types. See the information to the right for a full list of `oraclize_getPriceERC20` function overloads. All overloaded versions of the `oraclize_getPrice` helper function have the `view` visibility specifier and so are free for your contract to call.
+
+<aside class="notice">
+Note that the first query to the Provable service is _free_, and so the first call to `oraclize_getPriceERC20` from any contract with a gas limit and/or gas price that is _less than or equal_ to the default values will result in the function _correctly_ returning `0`.
+</aside>
 
-To enable the ProofShield it is enough to set it via the `oraclize_setProof` function like you see in the following code:
+In the case of a contract specifying a token payment, but sending enough ETH to cover the cost of the query, ETH will be used as payment rather than the token.
```

Comment:
ah, this is interesting and i do not recall it

Reply:
Yup: __[behaviour tested here.](https://github.com/oraclize/connector-update-priv/blob/339db47b9c16aca36a59ce40c61948f7352443e8/test/connector/making-queries.js#L975)__.

Reply:
kinda like having our facebook status saying: 

> Looking for: Whatever I can get